src/context/EarnManagerProvider/providers/yearn/components/YearnManager/YearnDeposit.tsx

```diff
@@ -69,46 +148,100 @@ export const YearnDeposit = ({ api }: YearnDepositProps) => {
   const location = useLocation()
   const depositRoute = matchPath(location.pathname, { path: DepositPath.Deposit, exact: true })
 
-  useEffect(() => {
-    ;(async () => {
-      if (!walletState.wallet) return
-      const chainAdapter = chainAdapterManager.byChain(ChainTypes.Ethereum)
-      setUserAddress(await chainAdapter.getAddress({ wallet: walletState.wallet }))
-    })()
-  }, [chain, chainAdapterManager, walletState])
-
   useEffect(() => {
     ;(async () => {
       try {
-        const apy = await api.apy({ vaultAddress })
-        setApy(apy)
+        if (!walletState.wallet) return
+        const [address, apy] = await Promise.all([
+          chainAdapter.getAddress({ wallet: walletState.wallet }),
+          api.apy({ vaultAddress })
+        ])
+        dispatch({ type: YearnActionType.SET_USER_ADDRESS, payload: address })
+        dispatch({ type: YearnActionType.SET_VAULT, payload: { apy } })
       } catch (error) {
-        // TODO: handle client side errors
```

Comment:
I don't think we should remove this TODO unless there really isn't anything to do, like show a toast or something?